tower-hedge/src/lib.rs

```diff
@@ -0,0 +1,199 @@
+extern crate futures;
+extern crate hdrhistogram;
+#[macro_use]
+extern crate log;
+extern crate tokio_timer;
+extern crate tower_filter;
+extern crate tower_service;
+
+use futures::future;
+use futures::future::FutureResult;
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+use tower_filter::Filter;
+
+mod delay;
+mod latency;
+mod rotating_histogram;
+mod select;
+
+use delay::Delay;
+use latency::Latency;
+use rotating_histogram::RotatingHistogram;
+use select::Select;
+
+type Histo = Arc<Mutex<RotatingHistogram>>;
+pub type Service<InnerSvc, P> = select::Select<
```

Comment:
The current convention is to name the service after the function (ideally a verb), so in this case, `Hedge` and to make it a type that contains the inner composition. So:

```rust
pub struct Hedge<T, P> {
    inner: Service<T, P>, // where `Service` is the current alias
```